packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte

```diff
@@ -165,11 +168,20 @@
     editableColumn?.type !== AUTO_TYPE &&
     editableColumn?.type !== JSON_TYPE &&
     !editableColumn.autocolumn
+  $: canHaveDefault =
+    isEnabled(TENANT_FEATURE_FLAGS.DEFAULT_VALUES) &&
+    (editableColumn?.type === FieldType.NUMBER ||
```

Comment:
Is this controlled only via frontend? Don't we have a backend validation?

Reply:
NAB: could also refactor into `[FieldType.NUMBER, FieldType...].includes(editableColumn?.type)`

Reply:
Currently only controlled via the frontend. Should be easy to add backend validation if you thought it was worth it, but at the time I felt like it was easier to reason about this only being toggled in the frontend.

Reply:
@melohagan In an ideal would I'd like to do a broader refactor of this file to rely far less on checking hardcoded field types, and instead have the FieldSchema dictate what can and cannot be done with a field, then genericise this component using the FieldSchema data.

Reply:
Just worried of having inconsistencies if these values are set somehow. It it's not too hard, having a helper in shared-core that is used both in front and back would be great

Reply:
Have had a go at what I think you're looking for, PTAL 🙂